### PR TITLE
Use a single GitHub bot client throughout the code

### PR DESCRIPTION
**Pull request by Hexcles**
Based on #2160

### CODE REVIEW

**Review by KyleJu — COMMENTED** [review]
LGTM on the code; since we are combining the GitHub clients, let me look at the request limit (5000/hour) on the triage metadata side, as well as other parts that use GitHub APIs

**Comment by Hexcles** [comment]
Do you have a pointer to your original estimation of API usage?

Currently almost everywhere uses `github-api-token`, including fetching manifest and metadata. The only place that uses `github-wpt-fyi-bot-token` is
https://github.com/web-platform-tests/wpt.fyi/blob/b8b8106cd7320a71f628214803cf583bde53f231/api/metadata_handler.go#L54
which is responsible for checking user membership and creating the metadata PRs.

In other words, these API uses don't seem to overlap so they shouldn't share the same quota IIUC.

**Comment by KyleJu** [comment]
> Do you have a pointer to your original estimation of API usage?
> 
> Currently almost everywhere uses `github-api-token`, including fetching manifest and metadata. The only place that uses `github-wpt-fyi-bot-token` is
> https://github.com/web-platform-tests/wpt.fyi/blob/b8b8106cd7320a71f628214803cf583bde53f231/api/metadata_handler.go#L54
> 
> 
> which is responsible for checking user membership and creating the metadata PRs.
> In other words, these API uses don't seem to overlap so they shouldn't share the same quota IIUC.

I think the quota applies to all API calls that an authenticated user requests in an hour. That being said, if it never exceeded for `github-api-token`, there shouldn't be a problem. It requires 8 API calls per triage (including checking membership/creating a PR), and at peak there are at most 15 PRs triaged in an hour, which brings 120 calls.

**Review by KyleJu — APPROVED** [review]

File: api/metadata_handler.go [hunk 1]
```diff
@@ -30,11 +30,13 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	ctx := shared.NewAppEngineContext(r)
-	client := shared.NewAppEngineAPI(ctx).GetHTTPClient()
+	aeAPI := shared.NewAppEngineAPI(ctx)
 	logger := shared.GetLogger(ctx)
+	client := aeAPI.GetHTTPClient()
 	gitHubClient, err := shared.NewAppEngineAPI(ctx).GetGitHubClient()
```
Comment (Hexcles) [inline api/metadata_handler.go:36]:
```suggestion
	gitHubClient, err := aeAPI.GetGitHubClient()
```

File: api/metadata_handler.go [hunk 2]
```diff
@@ -45,20 +47,22 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
-	botToken, err := shared.GetSecret(ds, "github-wpt-fyi-bot-token")
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	logger := shared.GetLogger(ctx)
+	botClient, err := aeAPI.GetGitHubClient()
 	if err != nil {
-		http.Error(w, "Unable to get GitHub client for wpt-fyi-bot: "+err.Error(), http.StatusInternalServerError)
+    logger.Errorf("Unable to get GitHub client: %s", err.Error())
```
Comment (Hexcles) [inline api/metadata_handler.go:61]:
```suggestion
	 	 logger.Errorf("Unable to get GitHub client: %s", err.Error())
```

File: api/metadata_handler.go [hunk 3]
```diff
@@ -68,7 +72,8 @@ func apiMetadataTriageHandler(w http.ResponseWriter, r *http.Request) {
 
 	gac, err := shared.NewGitHubAccessControl(ctx, ds, botClient, user, token)
 	if err != nil {
-		http.Error(w, "Unable to create GitHub OAuth client: "+err.Error(), http.StatusInternalServerError)
+    logger.Errorf("Unable to create GitHub OAuth client: %s", err.Error())
```
Comment (Hexcles) [inline api/metadata_handler.go:75]:
```suggestion
	 	 logger.Errorf("Unable to create GitHub OAuth client: %s", err.Error())
```

File: api/metadata_handler.go [hunk 4]
```diff
@@ -45,20 +47,22 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
-	botToken, err := shared.GetSecret(ds, "github-wpt-fyi-bot-token")
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	logger := shared.GetLogger(ctx)
+	botClient, err := aeAPI.GetGitHubClient()
 	if err != nil {
-		http.Error(w, "Unable to get GitHub client for wpt-fyi-bot: "+err.Error(), http.StatusInternalServerError)
+	 	 logger.Errorf("Unable to get GitHub client: %s", err.Error())
```
Comment (Hexcles) [inline api/metadata_handler.go:61]:
```suggestion
		logger.Errorf("Unable to get GitHub client: %s", err.Error())
```

File: api/metadata_handler.go [hunk 5]
```diff
@@ -68,7 +72,8 @@ func apiMetadataTriageHandler(w http.ResponseWriter, r *http.Request) {
 
 	gac, err := shared.NewGitHubAccessControl(ctx, ds, botClient, user, token)
 	if err != nil {
-		http.Error(w, "Unable to create GitHub OAuth client: "+err.Error(), http.StatusInternalServerError)
+	 	 logger.Errorf("Unable to create GitHub OAuth client: %s", err.Error())
```
Comment (Hexcles) [inline api/metadata_handler.go:75]:
```suggestion
		logger.Errorf("Unable to create GitHub OAuth client: %s", err.Error())
```

File: api/metadata_handler.go [hunk 6]
```diff
@@ -30,11 +30,13 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
...
 
 	ctx := shared.NewAppEngineContext(r)
-	client := shared.NewAppEngineAPI(ctx).GetHTTPClient()
+	aeAPI := shared.NewAppEngineAPI(ctx)
 	logger := shared.GetLogger(ctx)
+	client := aeAPI.GetHTTPClient()
 	gitHubClient, err := shared.NewAppEngineAPI(ctx).GetGitHubClient()
 	if err != nil {
-		http.Error(w, "Unable to get GitHub client: "+err.Error(), http.StatusInternalServerError)
+		logger.Errorf("Unable to get GitHub client: %e", err)
```
Comment (Hexcles) [inline api/metadata_handler.go:38]:
```suggestion
		logger.Errorf("Unable to get GitHub client: %s", err.Error())
```

File: api/metadata_handler.go [hunk 7]
```diff
@@ -45,20 +47,22 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
-	botToken, err := shared.GetSecret(ds, "github-wpt-fyi-bot-token")
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	logger := shared.GetLogger(ctx)
+	botClient, err := aeAPI.GetGitHubClient()
 	if err != nil {
-		http.Error(w, "Unable to get GitHub client for wpt-fyi-bot: "+err.Error(), http.StatusInternalServerError)
+		logger.Errorf("Unable to get GitHub client: %e", err)
```
Comment (Hexcles) [inline api/metadata_handler.go:61]:
```suggestion
		logger.Errorf("Unable to get GitHub client: %s", err.Error())
```
